File: Lab2/WinFormsApp1/WinFormsApp1/Form1.vb
Public Class Form1
    Public Sub New()
        ' 9 VARIANT TASK 1
        InitializeComponent()

        AddHandler ButtonCalculate.Click, AddressOf ButtonCalculate_Click
    End Sub

    Private Sub ButtonCalculate_Click(sender As Object, e As EventArgs)
        LabelError.Text = String.Empty
        Dim R1, R2, R3, R4 As Double

        If Not Double.TryParse(TextBoxR1.Text, R1) OrElse R1 <= 0 OrElse
           Not Double.TryParse(TextBoxR2.Text, R2) OrElse R2 <= 0 OrElse
           Not Double.TryParse(TextBoxR3.Text, R3) OrElse R3 <= 0 OrElse
           Not Double.TryParse(TextBoxR4.Text, R4) OrElse R4 <= 0 Then
            LabelError.Text = "Введите корректные положительные значения для всех резисторов."
            Return
        End If

        Dim Rtotal As Double
        If RadioButtonSeries.Checked Then
            Rtotal = R1 + R2 + R3 + R4
        ElseIf RadioButtonParallel.Checked Then
            Rtotal = 1 / (1 / R1 + 1 / R2 + 1 / R3 + 1 / R4)
        Else
            LabelError.Text = "Выберите тип соединения."
            Return
        End If

        LabelResult.Text = $"Общее сопротивление: {Rtotal:F2} Ом"
    End Sub
End Class

File: Lab2/Lab2_Task1/Lab2_Task1/Form1.cs
using System;
using System.Windows.Forms;

namespace Lab2_Task1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            button1.Click += ButtonCalculate_Click;
        }

        private void ButtonCalculate_Click(object sender, EventArgs e)
        {
            LabelError.Text = string.Empty;
            double R1, R2, R3, R4;

            if (!double.TryParse(textBox1.Text, out R1) || R1 <= 0 ||
                !double.TryParse(textBox2.Text, out R2) || R2 <= 0 ||
                !double.TryParse(textBox3.Text, out R3) || R3 <= 0 ||
                !double.TryParse(textBox4.Text, out R4) || R4 <= 0)
            {
                LabelError.Text = "Enter correct positive values for all resistors.";
                return;
            }

            double Rtotal;
            if (RadioButtonSeries.Checked)
            {
                Rtotal = R1 + R2 + R3 + R4;
            }
            else if (RadioButtonParallel.Checked)
            {
                Rtotal = 1 / (1 / R1 + 1 / R2 + 1 / R3 + 1 / R4);
            }
            else
            {
                LabelError.Text = "Select the connection type.";
                return;
            }

            LabelResult.Text = $"Total resistance: {Rtotal:F2} Ohms";
        }
    }
}
